File: Java_Solutions/Solution1.java
public class Solution1 {
    // User input
    public static int sum(int a, int b) {
        return a + b;
    }
    // End of user input

    static class Checker {
        public static void check(int a, int b, int expected) {
            int result = sum(a, b);
            if (result == expected) {
                System.out.printf("Test Case Passed: %d + %d = %d%n", a, b, expected);
            } else {
                System.out.printf("Failed Test Case: %d + %d != %d%n", a, b, expected);
            }
        }
    }

    public static void main(String[] args) {
        if (args.length != 3) {
            System.err.printf("Usage: java Solution1 <a> <b> <expected>%n");
            System.exit(1);
        }

        int a = Integer.parseInt(args[0]);
        int b = Integer.parseInt(args[1]);
        int expected = Integer.parseInt(args[2]);

        Checker.check(a, b, expected);
    }
} 

File: Solutions/Rust_Solutions/solution1.rs
use std::env;

// User input
fn sum(a: i32, b: i32) -> i32 {
    a + b
}
// End of user input

struct Checker;

impl Checker {
    fn check(func: fn(i32, i32) -> i32, a: i32, b: i32, expected: i32) {
        let result = func(a, b);
        if result == expected {
            println!("Test Case Passed: {} + {} = {}", a, b, expected);
        } else {
            println!("Failed Test Case: {} + {} != {}", a, b, expected);
        }
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 4 {
        eprintln!("Usage: {} <a> <b> <expected>", args[0]);
        std::process::exit(1);
    }

    let a: i32 = args[1].parse().expect("Invalid first number");
    let b: i32 = args[2].parse().expect("Invalid second number");
    let expected: i32 = args[3].parse().expect("Invalid expected value");

    Checker::check(sum, a, b, expected);
} 